src/script_lua.c

```diff
@@ -1292,7 +1292,12 @@ void luaCallFunction(scriptRunCtx* run_ctx, lua_State *lua, robj** keys, size_t
     /* At this point whether this script was never seen before or if it was
      * already defined, we can call it. We have zero arguments and expect
      * a single return value. */
-    int err = lua_pcall(lua,run_ctx->flags & SCRIPT_EVAL_MODE ? 0 : 2,1,run_ctx->flags & SCRIPT_EVAL_MODE ? -2 : -4);
+    int err;
+    if (run_ctx->flags & SCRIPT_EVAL_MODE) {
+        err = lua_pcall(lua,0,1,-2);
+    } else {
+        err = lua_pcall(lua,2,1,-4);
```

Comment:
let's add a comment explaining the difference between the two

Reply:
Fixed